File: android/sdui-core/src/main/java/com/nba/sdui/core/network/WireUrlResolver.kt
package com.nba.sdui.core.network

/**
 * Resolves server-emitted relative asset paths (e.g. `/sdui-demo/card-wide.svg`)
 * against the SDUI composition service [baseUrl] used by [com.nba.sdui.core.data.SduiRepository].
 *
 * Web serves the same paths same-origin via the dev proxy; Android/iOS must
 * absolutize before handing URLs to Coil / Kingfisher.
 */
object WireUrlResolver {
    fun resolve(url: String?, baseUrl: String?): String? {
        if (url.isNullOrBlank()) return url
        val trimmed = url.trim()
        if (trimmed.startsWith("http://", ignoreCase = true) ||
            trimmed.startsWith("https://", ignoreCase = true)
        ) {
            return trimmed
        }
        val base = baseUrl?.trim()?.trimEnd('/') ?: return trimmed
        val path = if (trimmed.startsWith("/")) trimmed else "/$trimmed"
        return base + path
    }
}

File: ios/Sources/SduiCore/Network/WireUrlResolver.swift
import Foundation

/// Resolves server-emitted relative asset paths (e.g. `/sdui-demo/team.svg`)
/// against the SDUI composition service base URL used by ``SduiRepository``.
///
/// Web serves the same paths same-origin via the dev proxy; iOS must
/// absolutize before handing URLs to Kingfisher.
enum WireUrlResolver {
    static func resolve(_ url: String?, baseURL: String?) -> String? {
        guard let url else { return nil }
        let trimmed = url.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !trimmed.isEmpty else { return trimmed }
        let lower = trimmed.lowercased()
        if lower.hasPrefix("http://") || lower.hasPrefix("https://") {
            return trimmed
        }
        var base = baseURL?.trimmingCharacters(in: .whitespacesAndNewlines) ?? ""
        while base.hasSuffix("/") { base.removeLast() }
        guard !base.isEmpty else { return trimmed }
        let path = trimmed.hasPrefix("/") ? trimmed : "/\(trimmed)"
        return base + path
    }
}
